src/main/scala/io/iohk/ethereum/network/PeerStatisticsActor.scala

```diff
@@ -0,0 +1,113 @@
+package io.iohk.ethereum.network
+
+import akka.actor._
+import cats._
+import cats.implicits._
+import io.iohk.ethereum.network.PeerEventBusActor._
+import io.iohk.ethereum.network.p2p.messages.Codes
+import java.time.Clock
+import scala.concurrent.duration.FiniteDuration
+import cats.kernel.Monoid
+
+class PeerStatisticsActor(
+    peerEventBus: ActorRef,
+    var maybeStats: Option[TimeSlotStats[PeerId, PeerStatisticsActor.Stat]]
+) extends Actor {
+  import PeerStatisticsActor._
+
+  // Subscribe to messages received from handshaked peers to maintain stats.
```

Comment:
Maybe we can encapsulated calls to peerEventBus actor

override def preStart(): Unit = {
 self ! Initialization
}

private def handleInitilization: Receive = {
 // Subscribe to messages received from handshaked peers to maintain stats.
 peerEventBus ! Subscribe(MessageSubscriptionClassifier)
 // Removing peers is an optimisation to free space, but eventually the stats would be overwritten anyway.
 peerEventBus ! Subscribe(SubscriptionClassifier.PeerDisconnectedClassifier(PeerSelector.AllPeers))
}

What do you think?

Reply:
I just followed what's happening when the [PeerManagerActor subscribes](https://github.com/input-output-hk/mantis/blob/develop/src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala#L71-L72). 

The `preStart` looks cleaner than a side effecting constructor body, but why not send directly to the event bus from `preStart`, why the extra indirection?

Reply:
I see [constructor](https://github.com/input-output-hk/mantis/blob/develop/src/main/scala/io/iohk/ethereum/transactions/PendingTransactionsManager.scala#L87) and [preStart](https://github.com/input-output-hk/mantis/blob/develop/src/main/scala/io/iohk/ethereum/blockchain/sync/PeerRequestHandler.scala#L37) both being used in the codebase for subscribing to the event bus. Strictly speaking the actor only needs to subscribe once, it wouldn't have to do it after restarts, which could be [achieved](https://stackoverflow.com/questions/51538451/constructor-vs-prestart-when-to-use-which) with `preStart` only. 

I'll move it into `preStart` :+1:

Reply:
Done. Didn't want the extra `Receive` because then I'd have to think about whether to keep expecting it forever, or use `context.become` and it seems unnecessary.